### PR TITLE
feat: resume branches loops

### PR DESCRIPTION
**Pull request by khaledmashaly**
## What does this PR do?

Fixes # (issue)

### CODE REVIEW

File: packages/engine/src/lib/action/branch-action-handler.ts [hunk 1]
```diff
@@ -28,52 +35,72 @@ export class BranchActionHandler extends BaseActionHandler<BranchAction> {
...
-        }
       }
+
+      executionState.insertStep(stepOutput, this.currentAction.name, ancestors);
+
+      const firstStep = condition
+        ? this.onSuccessAction
+        : this.onFailureAction
+
+      if (isNil(firstStep)) {
```
Comment (abuaboud) [inline packages/engine/src/lib/action/branch-action-handler.ts]: OnSuccessAction / onFailureAction is optional field, so throwing an error would make the flow stop.

File: packages/shared/src/lib/flow-run/execution/step-output.ts [hunk 2]
```diff
@@ -11,6 +14,7 @@ export class StepOutput<T = any>{
   errorMessage?: unknown;
   standardOutput?: unknown;
   status?: StepOutputStatus;
+  pauseMetadata?: Omit<PauseMetadata, 'executionState'>
```
Comment (abuaboud) [inline packages/shared/src/lib/flow-run/execution/step-output.ts:19]: Why this should exists inside StepOutput? Maybe we can use the global in engine if we want to only the one that paused?

**Review by abuaboud — CHANGES_REQUESTED** [review]

File: packages/engine/src/lib/action/branch-action-handler.ts (hunk 1 above)
Comment (khaledmashaly) [reply packages/engine/src/lib/action/branch-action-handler.ts]: fixed in 0cf89088

File: packages/shared/src/lib/flow-run/execution/step-output.ts (hunk 2 above)
Comment (khaledmashaly) [reply packages/shared/src/lib/flow-run/execution/step-output.ts:19]: because it needs to be recursive

**Pull request merged**